posthog/queries/trends/trends_actors.py

```diff
@@ -10,27 +11,41 @@
 from posthog.models.entity import Entity
 from posthog.models.filters import Filter
 from posthog.models.filters.mixins.utils import cached_property
+from posthog.models.filters.properties_timeline_filter import PropertiesTimelineFilter
 from posthog.models.person.sql import GET_ACTORS_FROM_EVENT_QUERY
 from posthog.models.property import Property
 from posthog.models.team import Team
 from posthog.queries.actor_base_query import ActorBaseQuery
 from posthog.queries.trends.trends_event_query import TrendsEventQuery
 from posthog.queries.trends.util import PROPERTY_MATH_FUNCTIONS, is_series_group_based, process_math
 
+F = TypeVar("F", Filter, PropertiesTimelineFilter)
 
-def _handle_date_interval(filter: Filter) -> Filter:
-    # adhoc date handling. parsed differently with django orm
+
+def handle_date_to_with_interval_for_data_point_actors(filter: F, team: Team) -> F:
```

Comment:
This is kind of a mess and I have no way to judge whether this is correct or not.

Why do we need to expose this function to the world now? 

It's mutating immutable data in a really complicated way so exposing it to more places just makes working with any queries worse.

Suggestion: Only expose a function that returns updated date_from/date_to

Reply:
I agree this is a mess, I've just discovered it. 😄 The real fix would be making actors modal URLs sane by setting "fully qualified" (ISO string) `date_from` and `date_to` in them, instead of this mess where non-cumulative time-series insights have special logic.
And you know what, I think I'll just do that in a separate PR now. Things will be much more obvious if the dates are precomputed properly + better from a caching perspective. I'll retain this special "adhoc date handling" so that any cached insights work the same, but the faster we make the API change, the sooner we'll be able to clean this up completely.

Reply:
Yeah, let's do that, I'll remove `handle_date_to_with_interval_for_data_point_actors()`.

Reply:
Or well, actually will probably remove it in some later follow-up, since there will be cached insights with this annoying complexity for a longer time. But will limit the need for this to legacy `date_from == date_to` situations.